### PR TITLE
Recipe app

### PR DESCRIPTION
**Pull request by Roland-Ntwali**
## In this PR, we did the following:

- A login page.
- A registration page.
- A food list (with all CRUD implementation, except for 'update').
- A recipes list (with all CRUD implementation, except for 'update').
- A general shopping list view (all missing food for all your recipes and total price).
- A list of all public recipes from other users with their names and total prices.

### CODE REVIEW

File: app/views/foods/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    <tbody>
+        <% @foods.each do |food| %>
+        <tr>
+            <td><%= food.name %></td>
+            <td><%= food.measurement_unit %></td>
+            <td>$<%= food.price %></td>
+            <td>
+              <% if can? :manage, food %>
+              <%= button_to 'Delete', food_path(food), method: :delete, data: { turbo: false }, class: "delete row-delete" %>
+              <% end %>
```
Comment (fobadara) [inline app/views/foods/index.html.erb]: [REQUIRED] You have an extra column in your table which goes against the wireframe. Kindly fix this 
image

File: app/views/recipe_foods/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    <%= form.select :food_id, @foods.map { |food| [food.name, food.id] },
+      { include_blank: true }, class: "form-field" %>
+  </div>
+  <div>
+    <%= form.number_field :quantity, placeholder: 'Quantity', class: "form-field" %>
+  </div>
+
+  <div>
+    <%= form.submit "Add ingredient", class: "link-wrapper"%>
+  </div>
```
Comment (fobadara) [inline app/views/recipe_foods/new.html.erb:14]: [REQUIRED] Clicking on the add ingredient button doesn't add ingredient to the recipe

**Review by fobadara — CHANGES_REQUESTED** [review]
Hi team, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

## Highlights
✅ Good first attempt
✅ Clean code
✅ Use of strong params
✅ Use of includes when necessary



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/recipe_foods/new.html.erb (hunk 2 above)
Comment (rloterh) [reply app/views/recipe_foods/new.html.erb:14]: well resolved! ✔️

File: app/views/foods/index.html.erb (hunk 1 above)
Comment (rloterh) [reply app/views/foods/index.html.erb]: well resolved! ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Basim & Roland,  

Your project is complete! 🏁 Fantastic job! 💎💎 There is nothing else to say other than... it's time to merge it 🍾 Congratulations!  🎉💯🌟 

## Highlights 🎯 
🎲 Awesome job for introducing improvements to the app's design and functionality!
🎲 Authentication with Devise gem properly implemented!
🎲 App schema closely follows given ERD!
🎲 Awesome UI!
🎲 ReadMe looks great!

## Optional suggestions 🧰 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

_Kindly consider my inline comments marked **[OPTIONAL]** under this review_

Keep being awesome! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/recipe_foods/new.html.erb (hunk 2 above)
Comment (Roland-Ntwali) [reply app/views/recipe_foods/new.html.erb:14]: Great. Thank you!

**Pull request merged**